kura/org.eclipse.kura.nm/src/main/java/org/eclipse/kura/nm/configuration/NMConfigurationServiceImpl.java

```diff
@@ -295,6 +314,81 @@ private void decryptAndConvertPasswordProperties(Map<String, Object> modifiedPro
         }
     }
 
+    private void decryptAndConvertCertificatesProperties(Map<String, Object> modifiedProps, Set<String> interfaces) {
+
+        interfaces.forEach(interfaceName -> {
+            String key = String.format("net.interface.%s.config.802-1x.keystore.pid", interfaceName);
+            if (modifiedProps.containsKey(key)) {
+                
+                Object prop = modifiedProps.get(key);
+
+                if (prop instanceof String) {
+                    String keystorePid = (String) prop;
+    
+                    findAndDecodeCertificatesForInterface(interfaceName, modifiedProps,
+                            this.keystoreServices.get(keystorePid));
+                }
+            }
+        });
+    }
+
+    private void findAndDecodeCertificatesForInterface(String interfaceName, Map<String, Object> modifiedProps,
+            KeystoreService keystoreService) {
+
+        if (keystoreService == null) {
+            logger.error("Cannot find keystore service for interface {}", interfaceName);
+            return;
+        }
+
+        final String clientCertString = String.format("net.interface.%s.config.802-1x.client-cert-name", interfaceName);
+        final String caCertString = String.format("net.interface.%s.config.802-1x.802-1x.ca-cert-name", interfaceName);
+        final String privateKeyString = String.format("net.interface.%s.config.802-1x.private-key-name", interfaceName);
+        final List<String> keyCertStrings = Arrays.asList(clientCertString, caCertString, privateKeyString);
+
+        for (String key : keyCertStrings) {
+            if (!modifiedProps.containsKey(key)) {
+                continue;
+            }
+
+            Object value = modifiedProps.get(key);
+            try {
+                String valueString = value.toString();
+                if (isCertificate(key)) {
+                    modifiedProps.put(key, getTrustedCertificateFromKeystore(valueString, keystoreService));
+                } else {
+                    modifiedProps.put(key, getTrustedPrivateKeyFromKeystore(valueString, keystoreService));
+                }
+            } catch (KuraException e) {
+                logger.error("Unable to decode key/certificate {} from keystore.", key, e);
+                modifiedProps.put(key, value);
+            }
+        }
+    }
+
+    private boolean isCertificate(String key) {
+        return key.contains("cert");
+    }
```

Comment:
I know it works and it's ok as-is but I would further future proof it by being a little bit more specific, just to be on the safe side:

```suggestion
    private boolean isCertificate(String key) {
        return key.contains("802-1x.client-cert-name") || key.contains("802-1x.ca-cert-name");
    }
```